admin/provisioner/resources.go

```diff
@@ -91,11 +92,13 @@ func NewClickhouseConfig(cfg map[string]any) (*ClickhouseConfig, error) {
 	return res, nil
 }
 
-func (r *ClickhouseConfig) AsMap() map[string]any {
+func (c *ClickhouseConfig) AsMap() map[string]any {
 	res := make(map[string]any)
-	err := mapstructure.Decode(r, &res)
-	if err != nil {
-		panic(err)
+	if c.DSN != "" {
+		res["dsn"] = c.DSN
+	}
+	if c.WriteDSN != "" {
+		res["write_dsn"] = c.WriteDSN
 	}
```

Comment:
Maybe keep the `mapstructure` and add `omitempty` in the struct tags instead? https://pkg.go.dev/github.com/mitchellh/mapstructure#hdr-Omit_Empty_Values